include/aws/common/allocator.h

```diff
@@ -0,0 +1,110 @@
+#ifndef AWS_COMMON_ALLOCATOR_H
+#define AWS_COMMON_ALLOCATOR_H
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/common/macros.h>
+#include <aws/common/stdint.h>
+
+AWS_EXTERN_C_BEGIN
+
+/* Allocator structure. An instance of this will be passed around for anything needing memory allocation */
+struct aws_allocator {
+    void *(*mem_acquire)(struct aws_allocator *allocator, size_t size);
+    void (*mem_release)(struct aws_allocator *allocator, void *ptr);
+    /* Optional method; if not supported, this pointer must be NULL */
+    void *(*mem_realloc)(struct aws_allocator *allocator, void *oldptr, size_t oldsize, size_t newsize);
+    /* Optional method; if not supported, this pointer must be NULL */
+    void *(*mem_calloc)(struct aws_allocator *allocator, size_t num, size_t size);
+    void *impl;
+};
+
+/* Avoid pulling in CoreFoundation headers in a header file. */
+#ifdef __MACH__
+struct __CFAllocator;
+typedef const struct __CFAllocator *CFAllocatorRef;
```

Comment:
I think you can move this into the #ifdef block below, which should make it clearer to read.

Reply:
Done